File: backend/task-service/src/main/java/com/crm/task/model/Task.java
package com.crm.task.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "crm_tasks")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(name = "task_type", nullable = false, length = 50)
    @Builder.Default
    private TaskType taskType = TaskType.FOLLOW_UP;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    @Builder.Default
    private TaskPriority priority = TaskPriority.MEDIUM;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    @Builder.Default
    private TaskStatus status = TaskStatus.TODO;

    @Column(name = "due_date")
    private LocalDate dueDate;

    @Column(name = "completed_at")
    private LocalDateTime completedAt;

    @Column(name = "assigned_to_user_id")
    private Long assignedToUserId;

    @Column(name = "assigned_to_user_name", length = 120)
    private String assignedToUserName;

    @Enumerated(EnumType.STRING)
    @Column(name = "related_entity_type", length = 50)
    @Builder.Default
    private RelatedEntityType relatedEntityType = RelatedEntityType.GENERAL;

    @Column(name = "related_entity_id")
    private Long relatedEntityId;

    @Column(name = "related_entity_name", length = 150)
    private String relatedEntityName;

    @Column(name = "product_id")
    private Long productId;

    @Column(name = "product_name", length = 200)
    private String productName;

    @Column(name = "created_by_user_id")
    private Long createdByUserId;

    @Column(name = "created_by_user_name", length = 150)
    private String createdByUserName;

    @Column(name = "created_by_role", length = 50)
    private String createdByRole;

    @Column(name = "is_deleted", nullable = false)
    @Builder.Default
    private Boolean isDeleted = false;

    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: frontend/src/types/task.ts
export type TaskStatus = 'TODO' | 'IN_PROGRESS' | 'COMPLETED' | 'CANCELLED';

export type TaskPriority = 'LOW' | 'MEDIUM' | 'HIGH' | 'URGENT';

export type TaskType = 
  | 'FOLLOW_UP'
  | 'CALL'
  | 'MEETING'
  | 'EMAIL'
  | 'PROPOSAL'
  | 'ONBOARDING'
  | 'OTHER';

export type RelatedEntityType = 'LEAD' | 'CUSTOMER' | 'GENERAL';

export interface Task {
  id: number;
  title: string;
  description?: string;
  taskType: TaskType;
  taskTypeDisplayName: string;
  priority: TaskPriority;
  priorityDisplayName: string;
  status: TaskStatus;
  statusDisplayName: string;
  dueDate?: string;
  completedAt?: string;
  assignedToUserId?: number;
  assignedToUserName?: string;
  relatedEntityType: RelatedEntityType;
  relatedEntityDisplayName: string;
  relatedEntityId?: number;
  relatedEntityName?: string;
  productId?: number;
  productName?: string;
  createdByUserId?: number;
  createdByUserName?: string;
  createdByRole?: string;
  isDeleted?: boolean;
  isOverdue: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateTaskRequest {
  title: string;
  description?: string;
  taskType?: TaskType;
  priority?: TaskPriority;
  status?: TaskStatus;
  dueDate?: string;
  assignedToUserId?: number;
  assignedToUserName?: string;
  relatedEntityType?: RelatedEntityType;
  relatedEntityId?: number;
  relatedEntityName?: string;
  productId?: number;
  productName?: string;
}

export interface UpdateTaskRequest {
  title: string;
  description?: string;
  taskType?: TaskType;
  priority?: TaskPriority;
  status?: TaskStatus;
  dueDate?: string;
  assignedToUserId?: number;
  assignedToUserName?: string;
  relatedEntityType?: RelatedEntityType;
  relatedEntityId?: number;
  relatedEntityName?: string;
  productId?: number;
  productName?: string;
}

export interface TaskStats {
  totalTasks: number;
  todoTasks: number;
  inProgressTasks: number;
  completedTasks: number;
  cancelledTasks: number;
  overdueTasks: number;
  completionRate: number;
  tasksByPriority: Record<string, number>;
  tasksByStatus: Record<string, number>;
  tasksByType: Record<string, number>;
}
